internal/analysis/dolos/engine.go

```diff
@@ -44,8 +62,28 @@ func (this *dolosEngine) IsAvailable() bool {
 	return docker.CanAccessDocker()
 }
 
-func (this *dolosEngine) ComputeFileSimilarity(paths [2]string, templatePath string, ctx context.Context) (*model.FileSimilarity, error) {
-	err := ensureImage()
+func parseEngineOptions(rawOptions model.OptionsMap) (*DolosEngineOptions, error) {
```

Comment:
I think this function can be generalized to include the functionality for JPlag as well.

A possible signature could be:
```
func parseEngineOptions[T any](rawOptions model.OptionsMap, defaultOptions T) (T, error) {
```

It would probably be able to be called like:
```
options, err := parseEngineOptions(rawOptions, GetDefaultDolosOptions())
```